src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -444,12 +447,16 @@ StorageInfo PostgreSQLReplicationHandler::loadFromSnapshot(postgres::Connection
     assertBlocksHaveEqualStructure(input->getPort().getHeader(), block_io.pipeline.getHeader(), "postgresql replica load from snapshot");
     block_io.pipeline.complete(Pipe(std::move(input)));
 
+    /// TODO: make a test when we fail in the middle of inserting data from source.
```

Comment:
Perhaps introducing a  fail point can make it easier to reproduce
https://github.com/ClickHouse/ClickHouse/blob/efb82a93c5fd365caf01a850cb90511153e9fa7d/src/Common/FailPoint.cpp#L41

Reply:
Wow, cool, did not know about that, thanks.
Though this comment can be removed, because a test to which this comment is referring was added in https://github.com/ClickHouse/ClickHouse/pull/57567.